### PR TITLE
Make testing scripts python3.x compatible

### PR DESCRIPTION
**Pull request by thunderboltsid**
Update run_tests/*.py to use six based isomorphisms.

### CODE REVIEW

File: tools/run_tests/run_build_statistics.py [hunk 1]
```diff
@@ -243,6 +243,6 @@ def _process_build(json_url, console_url):
     rows = [big_query_utils.make_row(build_number, build_result)]
     if not big_query_utils.insert_rows(bq, _PROJECT_ID, _DATASET_ID, build_name, 
                                        rows):
-      print '====> Error uploading result to bigquery.'
+      print('====> Error uploading result to bigquery.')
```
Comment (nathanielmanistaatgoogle) [inline tools/run_tests/run_build_statistics.py]: Missing `from __future__ import print_function` in this file (should be after module doc string and before "ordinary" imports; see elsewhere around the codebase for example to follow).

**Comment by thunderboltsid** [comment]
@nathanielmanistaatgoogle I'd say that should have been the case from very beginning. Unfortunately, I can't predict what that would result in. There might still be some 2.x only code lying around that I have not stumbled around. What's your take on it?

**Comment by nathanielmanistaatgoogle** [comment]
I think it's something about which we should care, and like anything about which we should care, [we should test for it](https://github.com/grpc/grpc/issues/10117) (or at least have our test infrastructure depend on it).

**Comment by nathanielmanistaatgoogle** [comment]
I also think we should change our `#!/usr/bin/env python2.7`s to `#!/usr/bin/env python` today so that it at least _looks_ like we aspire to be version-agnostic and contributors don't see a "2.7" and don't bother to report bugs to us.

... and then in nine months we should change to `#!/usr/bin/env python3` and drop support for Python 2 because our test infrastructure, being mostly a thing to ourselves and not for the larger world, might as well use the latest and greatest language implementation. At this point there's basically no reason to ever _choose_ to use Python 2 and "testing gRPC" is an isolated enough effort that it should be able to be moved to Python 3 without any disruption to gRPC-Python-the-library's support of both versions.

**Comment by thunderboltsid** [comment]
@nathanielmanistaatgoogle I agree with using python3.x for test system but I'm not sure if we should do away with python2.x altogether. It's a nice experience when the test system is version agnostic.

**Comment by nathanielmanistaatgoogle** [comment]
@thunderboltsid: try to imagine living in a substantially more perfect world nine months from now? 😛 One more willing to break with a flawed and problematic past?

**Comment by matt-kwong** [comment]
```
Traceback (most recent call last):
  File "tools/run_tests/run_tests_matrix.py", line 41, in <module>
    import python_utils.report_utils as report_utils
  File "/c/jenkins/workspace/gRPC_pr_win/tools/run_tests/python_utils/report_utils.py", line 43, in <module>
    import six
ImportError: No module named six
Build step 'Execute shell' marked b
```
Six is installed on the Windows workers, so I'm not sure why this is happening. Could the issue stem from `!/usr/bin/env python` not being present at the top of `report_utils.py`? Even if it isn't the reason for the issue, I think it would be nice to add it to the top anyways.

**Comment by nathanielmanistaatgoogle** [comment]
@matt-kwong: shebangs are only appropriate in files that are meant to be executed. Files like `report_utils.py` that are merely libraries to be used by other Python code should not be given shebangs.

**Comment by matt-kwong** [comment]
Good to know, disregard what I said earlier then.

**Comment by nathanielmanistaatgoogle** [comment]
Pre-merge test review:

Basic Tests Mac: [9804](https://github.com/grpc/grpc/issues/9804).
Bazel Full Build & Tests: Not yet in effect.
Sanity Tests (internal CI, ignore for now): Ignoring for now.

Woot!

**Pull request merged**